kalite/distributed/static/js/distributed/search/views.js

```diff
@@ -94,89 +46,28 @@ var AutoCompleteView = BaseView.extend({
             minLength: 3,
             appendTo: ".navbar-collapse",
             html: true,  // extension allows html-based labels
-            source: this.data_source,
+            source: window.Urls.search_api(window.sessionModel.get("CHANNEL")),
```

Comment:
Refactor to query an API endpoint, rather than load a huge pile of data into the frontend.